accord-core/src/main/java/accord/coordinate/Barrier.java

```diff
@@ -66,7 +66,7 @@ public class Barrier<S extends Seekables<?, ?>> extends AsyncResults.AbstractRes
     private final BarrierType barrierType;
 
     @VisibleForTesting
-    CoordinateSyncPoint<S> coordinateSyncPoint;
+    AsyncResult<SyncPoint<S>> coordinateSyncPoint;
```

Comment:
Do we need to make this type less specific?

Reply:
yes, as we may return a `accord.utils.async.AsyncResults.Immediate` with a failure.